tests/fixtures/bar-foos-bar-route.js

```diff
@@ -0,0 +1,6 @@
+Router.map(function() {
```

Comment:
@srvance I say the multiple removals should happen because it's actually nonsensical and invalid:

```javascript
  this.route('foos', function() {
    this.route('bar');
    this.route('bar');
  });

// route: foos.bar is actually _both_ of them.
```

This will eventually throw an error in route-recognizer (which was previously released though we've just backed it out).